### DISCUSSION TITLE
Welcome to EDF Discussions!

### DISCUSSION THREAD

**Discussion by LiorBanai** [LiorBanai/EDF#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by andrbru** [comment]
Hello,

I would like to use your code in our project and I have two questions:

1. The project is written to run in .NET Framework 4.7.1 and higner. Would it be possible to compile your project with .NET 4.6 also?
2. Do I understand correctly that your code defines the class "AnnotationSignal" and "TAL" according to the EDF+ documentation about TALs but does not implement methods for reading EDF+ annotations from a .edf file, only methods for savig them?

The method Reader.ReadAnnotationSignals() mentioned in a note in your code is not defined anywhere. I could not find any other project for C# handling of EDF files including EDF+ annotations and your project seems to be the closest available.

Thank you a lot for your possible reaction!

**Reply by LiorBanai** [reply]
Hi @andrbru  , welcome.
1. no problem adding 4.6,
2. I never needed to read the annotations but I can add if if needed.

I'll update you.
do you have an example file for me to test it?

**Reply by andrbru** [reply]
Thanks a lot for such a fast reply! Adding option to read annotations would be absolutely awesome!

I am enclosing a file annotations.edf which when opened in the Polyman contains only Annotations signal.

I use the Polyman programme mentioned on the EDF specs website as a testing reader. It is described as a software used often for EDF popularization so I believe it handles the EDF+ files exactly according to the official specs.

I am further enclosing the file template.edf which causes a "divide by zero" exception when opened using your code but can be with no problems opened in Polyman.
[EDFFiles.zip](https://github.com/LiorBanai/EDF/files/9275210/EDFFiles.zip)

**Reply by LiorBanai** [reply]
Ok, I'll check it in the coming days and will update you

**Reply by LiorBanai** [reply]
For now, I have fixed the divide by zero exception since it is very easy (not sure how easy it will be to read the annotations themselves).

**Reply by andrbru** [reply]
Hi,

could you please publish/send me the version with the devide by zero exception fixed? And how does it look like with implementing annotations reading?

Thanks,
andrbru

**Reply by LiorBanai** [reply]
Hi @andrbru ,
the divide by zero is already fixed with v0.5.6
https://www.nuget.org/packages/EDF-CSharp/0.5.6

for the annotations I need more time (have limited time as I started new job two weeks ago)

**Reply by andrbru** [reply]
Sorry, I didn't check it. 

Thanks a lot -- especially for the time you are spending outside your job working on this.

If you may once have any idea of how long you might need I will be happy to be updated. But I understand that it's always hard to guess.

**Reply by LiorBanai** [reply]
I hope to have something by Monday.  will update

**Comment by LiorBanai** [comment]
Hi @andrbru ,
I have released new version at https://www.nuget.org/packages/EDF-CSharp/0.6.0
in this version I added support for reading the annotation signal entries.
give it a try and if there are issue please submit them as issues

**Reply by andrbru** [reply]
Hi,

Oh, that was fast! I will tinker with it during this weekend and I'll let you know if I find something.

I was not accualy using NuGet since now, just directly the code from GitHub, as it firstly didn't have the matching .NET Framework and as I can therefore look in your code, which helps me better understand which property or method does what and so... :grin: Do you think it may cause any complications? While since now it has worked well.

Thanks a lot for updating your library!

**Reply by LiorBanai** [reply]
well nuget supports debugging nuget packages so in theory you can debug it from visual studio but yes for looking at the code it is easier to just download/ clone the repo

**Comment by andrbru** [comment]
Just a simple feature idea -- wouldn't it be useful to make the FixedLengthString class control not only the minimum ASCII size of the field, but also the maximum? And in the same manner probably even VariableLengthString. E. g. simply change in the property Value the set-method so it checks if the value is not greater than AsciiLength and makes a substring if necessary.

I thought the class checks that automatically and get a bit suprised today learning they don't and one can therefore easily end up creating unreadable EDF files. 😅
